core/src/main/java/com/datastrato/gravitino/catalog/TopicOperationDispatcher.java

```diff
@@ -0,0 +1,267 @@
+import com.datastrato.gravitino.exceptions.TopicAlreadyExistsException;
+import com.datastrato.gravitino.messaging.DataLayout;
+import com.datastrato.gravitino.messaging.Topic;
+import com.datastrato.gravitino.messaging.TopicCatalog;
+import com.datastrato.gravitino.messaging.TopicChange;
+import com.datastrato.gravitino.meta.AuditInfo;
+import com.datastrato.gravitino.meta.TopicEntity;
+import com.datastrato.gravitino.storage.IdGenerator;
+import com.datastrato.gravitino.utils.PrincipalUtils;
+import java.time.Instant;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TopicOperationDispatcher extends OperationDispatcher implements TopicCatalog {
+  private static final Logger LOG = LoggerFactory.getLogger(TopicOperationDispatcher.class);
+
+  /**
+   * Creates a new TopicOperationDispatcher instance.
+   *
+   * @param catalogManager The CatalogManager instance to be used for catalog operations.
+   * @param store The EntityStore instance to be used for catalog operations.
+   * @param idGenerator The IdGenerator instance to be used for catalog operations.
+   */
+  public TopicOperationDispatcher(
+      CatalogManager catalogManager, EntityStore store, IdGenerator idGenerator) {
+    super(catalogManager, store, idGenerator);
+  }
+
+  /**
+   * List the topics in a schema namespace from the catalog.
+   *
+   * @param namespace A schema namespace.
+   * @return An array of topic identifiers in the namespace.
+   * @throws NoSuchSchemaException If the schema does not exist.
+   */
+  @Override
+  public NameIdentifier[] listTopics(Namespace namespace) throws NoSuchSchemaException {
+    return doWithCatalog(
+        getCatalogIdentifier(NameIdentifier.of(namespace.levels())),
+        c -> c.doWithTopicOps(t -> t.listTopics(namespace)),
+        NoSuchSchemaException.class);
+  }
+
+  /**
+   * Load topic metadata by {@link NameIdentifier} from the catalog.
+   *
+   * @param ident A topic identifier.
+   * @return The topic metadata.
+   * @throws NoSuchTopicException If the topic does not exist.
+   */
+  @Override
+  public Topic loadTopic(NameIdentifier ident) throws NoSuchTopicException {
+    NameIdentifier catalogIdent = getCatalogIdentifier(ident);
+    Topic topic =
+        doWithCatalog(
+            catalogIdent,
+            c -> c.doWithTopicOps(t -> t.loadTopic(ident)),
+            NoSuchTopicException.class);
+
+    StringIdentifier stringId = getStringIdFromProperties(topic.properties());
+    // Case 1: The topic is not created by Gravitino.
+    // Note: for Kafka catalog, stringId will not be null. Because there is no way to store the
+    // Gravitino
+    // ID in Kafka, therefor we use the topic ID as the Gravitino ID
+    if (stringId == null) {
+      return EntityCombinedTopic.of(topic)
+          .withHiddenPropertiesSet(
+              getHiddenPropertyNames(
+                  catalogIdent, HasPropertyMetadata::topicPropertiesMetadata, topic.properties()));
+    }
```

Comment:
This should never happened, right?

Reply:
As of now, yes. 

But if other messaging catalogs that support setting topic property are implemented later, such as Pulsar, this `stringId` may be null.